docs/language_guide.md

```diff
@@ -340,7 +340,7 @@ for var i: Int in fixedLen {}
 for let value: String in dictionary {}
 ```
 ### If Statements
-The `if` statement has a single condition  i.e. `x == 2`. It executes a set of statements only if that condition evaluates to `true`
+The `if` statement has a single condition, which should be either a Boolean expression (i.e. `x == 2`), or a let statement (i.e. `let y: Int = 5`). It executes a set of statements only if that condition evaluates to `true`.
```

Comment:
This should be expanded a bit more. The current wording implies a let statement is a condition that can evaluate to `true`. The two should be separate.

Additionally although currently `if let y: Int = 5 { ... }` is valid, it will not be once the semantic analysis for this is done. The `let` binding in `if` only makes sense for optional unwrapping, as is the case in external calls. Maybe we can update the language guide with this already, see the external calls proposal for valid examples (i.e. using `call?`).

Reply:
That's right, I will update the guide again. However, as a matter of fact, let statements as expressions actually do evaluate to something under the current implementations (for example, let x: Int = 5 evaluates to Int). Again, this might be worth investigating in the future.